src/axom/primal/operators/detail/intersect_bezier_impl.hpp

```diff
@@ -0,0 +1,204 @@
+ * When the two line segments intersect, their intersection parameters
+ * are stored in output parameters \a s and \a t, respectively, such that,
+ * if \a p is the intersection point,
+ * \f$    p = a + s * (b-a) = c + t * (d-c).  \f$
+ *
+ * \note We consider both line segments to have half-open intervals.
+ * As such, the we do not consider the lines to intersect if they do so
+ * at the endpoints \a b or \d, respectively.
+ *
+ * \note This function does not properly handle collinear lines
+ */
+
+template < typename T, int NDIMS>
+bool intersect_2d_linear( const Point<T,NDIMS> &a, const Point<T,NDIMS> &b,
+                          const Point<T,NDIMS> &c, const Point<T,NDIMS> &d,
+                          T &s, T &t);
+
+
+
+
+//------------------------------ IMPLEMENTATIONS ------------------------------
+
+
+template < typename T, int NDIMS>
+bool intersect_bezier_curves( const BezierCurve< T, NDIMS>& c1,
+                              const BezierCurve< T, NDIMS>& c2,
+                              std::vector< T >& sp,
+                              std::vector< T >& tp,
+                              double sq_tol,
+                              int order1, int order2,
+                              double s_offset, double s_scale,
+                              double t_offset, double t_scale)
+{
+  using BCurve = BezierCurve< T, NDIMS>;
+
+  // Check bounding boxes to short-circuit the intersection
+  if(!intersect(c1.boundingBox(), c2.boundingBox()) )
+  {
+    return false;
+  }
+
+  bool foundIntersection = false;
+
+  if ( c1.isLinear(sq_tol) && c2.isLinear(sq_tol))
+  {
+    T s,t;
+    if(intersect_2d_linear(c1[0],c1[order1],c2[0],c2[order2],s,t))
+    {
+      sp.push_back(s_offset + s_scale * s);
+      tp.push_back(t_offset + t_scale * t);
+      foundIntersection = true;
+    }
+  }
+  else
+  {
+    constexpr double splitVal = 0.5;
+    constexpr double scaleFac = 0.5;
+
+    BCurve c3(order1);
+    BCurve c4(order1);
+    c1.split(splitVal,c3,c4);
+
+    s_scale *= scaleFac;
+
+    if (intersect_bezier_curves(c2,c3,tp,sp, sq_tol, order2, order1,
+                                t_offset, t_scale, s_offset, s_scale))
+    {
+      foundIntersection = true;
+
+
+    }
+    if (intersect_bezier_curves(c2,c4,tp,sp, sq_tol, order2, order1,
+                                t_offset, t_scale, s_offset + s_scale, s_scale))
+    {
+      foundIntersection = true;
+    }
+  }
+
+  return foundIntersection;
+
+}
+
+
+template < typename T, int NDIMS>
+bool intersect_2d_linear( const Point<T,NDIMS> &a, const Point<T,NDIMS> &b,
+                          const Point<T,NDIMS> &c, const Point<T,NDIMS> &d,
+                          T &s, T &t)
+{
+  // Implementation inspired by Section 5.1.9.1 of
+  // C. Ericson's Real-Time Collision Detection book
+
+  // Note: Uses exact floating point comparisons since the subdivision algorithm
+  // provides both sides of the line segments for interior curve points.
```

Comment:
Note -- we are intentionally using exact floating point comparisons here, rather than fuzzy comparisons.  This prevents the subdivision algorithm from finding multiple occurrences of the same intersection.